073-decouple-ca-management-from-end-entity-certificate-management.md

```diff
@@ -0,0 +1,363 @@
+6. The EntityOperator, CruiseControl and KafkaExporter reconcilers generate new certificates signed by the new CA key and roll the Kubernetes Pods to use the new certificates.
+7. A new reconciliation starts.
+8. __The CaReconciler sees there is a CA certificate in the state TRUSTED_UNUSED in the trust-state Kubernetes Secret.
+   It checks if the cluster-ca-cert-generation annotation on the Kafka and ZooKeeper Kubernetes Pods matches the current CA cert generation.
+   If it does it updates the CA certificate state to TRUSTED_IN_USE, otherwise it rolls the Kafka and ZooKeeper Kubernetes Pods (jumping back to step 4 in this process).__
+9. __Since the CA certificate state is TRUSTED_IN_USE the CaReconciler updates the <CLUSTER_NAME>-cluster-operator-certs Kubernetes Secret with a new certificate signed by the new CA certificate.__
+10. __The CaReconciler removes the old Cluster CA certificate from the trusted-certs Kubernetes Secret and changes the state of it in the trust-state Kubernetes Secret to ZOMBIE.__
+11. __The CaReconciler rolls the Kafka and ZooKeeper nodes so they no longer trust the old Cluster CA.__
+12. __The CaReconciler removes the old Cluster CA certificate from the trust-state Kubernetes Secret.__
+13. The other component reconcilers (i.e. not Kafka or ZooKeeper) roll their Kubernetes Pods, since the Cluster CA passed in has a flag indicating certs have been removed.
+
+#### Fig 6: New CA certificate flow
+
+![Cluster CA private key update flow](./images/073-cert-update-proposed.png)
+
+> Fig 6: The numbers in the diagram match the numbers in the full description of the steps.
+> The diagram omits the use of annotations and the EntityOperator, CruiseControl and KafkaExporter reconcilers, and the removal of trust of the old CA certificate and key.
+
+Notes:
+* The existing annotations strimzi.io/ca-cert-generation and strimzi.io/server-cert-hash are still used to keep track of the certificate being presented by the Kubernetes Pod and which generation of the Cluster CA signed that certificate.
+* The strimzi.io/ca-key-generation is no longer needed, because it is being tracked in the trust-state Kubernetes Secret.
+
+### Future changes
+
+This proposal does not cover all changes needed to integrate with certificate management solutions.
+Instead it aims to make an improvement to the existing logic and act as a first step towards integrating some certificate management solution.
+The approach suggested aligns with the approach taken in the [closed proposal on CA abstraction][pr46].
+
+Other/Future proposals will be needed to cover:
+ - Refactoring the end-entity certificate issuance process, so that components request a certificate and asynchronously are issued a certificate.
+ - Refactoring the tracking of which certificates are signed by which CA certificates.
```

Comment:
Are you thinking about using the trust-state secret for that?

Reply:
Possibly, but I think it depends how we end up handling the asynchronous issuing of certificates. So until we implement this proposal and then look at how that can fit in I think it's unclear.